format/Format.py

```diff
@@ -317,6 +317,9 @@ def get_imageset(
         from dxtbx.imageset import ImageSequence, ImageSet, ImageSetData
 
         # Get filename absolute paths, for entries that are filenames
+        input_filenames = [
+            x.__fspath__() if hasattr(x, "__fspath__") else x for x in input_filenames
```

Comment:
```suggestion
            os.fspath(x) for x in input_filenames
```

Reply:
`x` can be a URL here, so am not sure this is appropriate here

Reply:
Ah, a URL object? Or just a string? because strings will be preserved....